File: shopapp-angular/src/app/manager/models/raw.product.ts
import { Provider } from 'src/app/admin/models/provider';

export interface RawProduct {
  id: number;
  name: string;
  quantity: number;
  note: string;
  price: number;
  created_at: Date;
  updated_at: Date;
  provider: Provider;
}

File: ShopAppSpring/src/main/java/com/duchung/shopappspring/models/RawProduct.java
package com.duchung.shopappspring.models;


import com.duchung.shopappspring.domains.BaseEntity;
import jakarta.persistence.*;
import lombok.*;

@Data
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "raw_products")
@Entity
@Builder
public class RawProduct extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 100)
    private String name;

    @Column(nullable = false)
    private double price;

    @Column(nullable = false)
    private Long quantity;

    @Column
    private String note;

    @ManyToOne
    @JoinColumn(name = "provider_id")
    private Provider provider;
}
